pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/operator/MailboxReceiveOperatorTest.java

```diff
@@ -0,0 +1,491 @@
+    TransferableBlock receivedBlock = receiveOp.nextBlock();
+    List<Object[]> resultRows = receivedBlock.getContainer();
+    Assert.assertEquals(resultRows.size(), 1);
+    Assert.assertEquals(resultRows.get(0), expRow1);
+    // Receive second block from first server.
+    receivedBlock = receiveOp.nextBlock();
+    resultRows = receivedBlock.getContainer();
+    Assert.assertEquals(resultRows.size(), 1);
+    Assert.assertEquals(resultRows.get(0), expRow2);
+
+    // Receive from second server.
+    receivedBlock = receiveOp.nextBlock();
+    resultRows = receivedBlock.getContainer();
+    Assert.assertEquals(resultRows.size(), 1);
+    Assert.assertEquals(resultRows.get(0), expRow3);
+  }
+
+  @Test
+  public void testReceiveMailboxFromTwoServersOneError()
+      throws Exception {
+    String server1Host = "hash1";
+    int server1Port = 123;
+    Mockito.when(_server1.getHostname()).thenReturn(server1Host);
+    Mockito.when(_server1.getQueryMailboxPort()).thenReturn(server1Port);
+
+    String server2Host = "hash2";
+    int server2Port = 456;
+    Mockito.when(_server2.getHostname()).thenReturn(server2Host);
+    Mockito.when(_server2.getQueryMailboxPort()).thenReturn(server2Port);
+
+    int jobId = 456;
+    int stageId = 0;
+    int toPort = 8888;
+    String toHost = "toHost";
+
+    DataSchema inSchema = new DataSchema(new String[]{"col1", "col2"}, new DataSchema.ColumnDataType[]{INT, INT});
+    StringMailboxIdentifier expectedMailboxId1 =
+        new StringMailboxIdentifier(String.format("%s_%s", jobId, stageId), server1Host, server1Port, toHost, toPort);
+    Mockito.when(_mailboxService.getReceivingMailbox(expectedMailboxId1)).thenReturn(_mailbox);
+    Mockito.when(_mailbox.isClosed()).thenReturn(false);
+    Mockito.when(_mailbox.receive())
+        .thenReturn(TransferableBlockUtils.getErrorTransferableBlock(new Exception("mailboxError")));
+
+    Object[] expRow3 = new Object[]{3, 3};
+    StringMailboxIdentifier expectedMailboxId2 =
+        new StringMailboxIdentifier(String.format("%s_%s", jobId, stageId), server2Host, server2Port, toHost, toPort);
+    Mockito.when(_mailboxService.getReceivingMailbox(expectedMailboxId2)).thenReturn(_mailbox2);
+    Mockito.when(_mailbox2.isClosed()).thenReturn(false);
+    Mockito.when(_mailbox2.receive()).thenReturn(OperatorTestUtil.block(inSchema, expRow3));
+    MailboxReceiveOperator receiveOp = new MailboxReceiveOperator(_mailboxService, ImmutableList.of(_server1, _server2),
+        RelDistribution.Type.HASH_DISTRIBUTED, toHost, toPort, jobId, stageId, null);
+    // Receive error block from first server.
+    TransferableBlock receivedBlock = receiveOp.nextBlock();
+    Assert.assertTrue(receivedBlock.isErrorBlock());
+    MetadataBlock error = (MetadataBlock) receivedBlock.getDataBlock();
+    Assert.assertEquals(error.getExceptions().get(QueryException.UNKNOWN_ERROR_CODE), "mailboxError");
+  }
+
+  // TODO: This should probably be fixed.
```

Comment:
can you explain what needs to be fixed?
did you mean the mocked exception should bubble up?